teps/0094-configuring-resources-at-runtime.md

```diff
@@ -0,0 +1,61 @@
+---
+status: proposed
+title: Configuring Resources at Runtime
+creation-date: '2021-11-08'
+last-updated: '2021-11-08'
+authors:
+- '@lbernick'
+---
+
+# TEP-0094: Configuring Resources at Runtime
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+<!-- /toc -->
+
+## Summary
+Add runtime configuration options for setting resource requirements of `Step`s and `Sidecar`s.
+
+Currently, users can specify resource requirements in a `Task` definition,
+via the `Resources` field of each `Step` or `Sidecar`. However, there is currently no support
+for modifying these requirements in a `TaskRun` or `PipelineTaskRun`.
+
+This TEP proposes adding a configuration option to `TaskRun` and `PipelineTaskRun`
+to override any `Step` or `Sidecar` resource requirements specified in a `Task` or `PipelineTask`.
+
+## Motivation
+Compute resource requirements typically depend on runtime constraints.
+The following issues contain user requests for being able to modify resource requirements at runtime:
+
+- [Allow usage of variable replacement when defining resource limits and requests](https://github.com/tektoncd/pipeline/issues/4080)
+- [Support specifying resource requests at TaskRun level](https://github.com/tektoncd/pipeline/issues/4326)
+
+### Goals
+
+Add configuration to `TaskRun` and `PipelineTaskRun` allowing users to specify resource requirements
+of `Step`s or `Sidecar`s defined in a `Task` or `PipelineTask`.
+
+### Non-Goals
+
+- Ability to override other `Step` or `Sidecar` fields in a `TaskRun` or `PipelineTaskRun`.
+- Ability to specify combined resource requirements of all `Step`s or `Sidecar`s at `Task` or `Pipeline` level.
+While this may be a valuable feature, it should be considered in a separate proposal.
+
+### Use Cases
+
+- Image or code building `Task`s can use different amounts of compute resources
+depending on the image or source being built.
```

Comment:
Maybe another use case is just generally reusing catalog Tasks (which i think https://github.com/tektoncd/pipeline/issues/4080 touched on) - we want items in the catalog to be reusable but the environment they are used in (at runtime) might have resource constraints that need to be expressed in the resulting pods